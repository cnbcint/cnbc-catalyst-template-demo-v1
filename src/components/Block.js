import React, { useContext } from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { breakpoints, getMargin } from "../helpers/Layout";
import { LayoutContext } from "./Main";

const Container = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  padding-top: ${getMargin(3)};
  padding-bottom: ${getMargin(3)};
  background-image: ${(props) =>
    props.backgroundImage ? "url(" + props.backgroundImage + ")" : "none"};
  @media (min-width: ${breakpoints.mobile}) {
    padding-top: ${getMargin(5)};
    padding-bottom: ${getMargin(5)};
  }
`;
const Content = styled.div`
  width: calc(100% - ${getMargin(2)});
  display: flex;
  flex-direction: column;
  max-width: ${(props) => props.maxWidth};
`;
/**
 *
 * Block is the main wrapper component for this template.
 * This element enables to have consistent spacing between sections using the theme functionality to define a central column
 * You can include most of the components
 */
export default function Block(props) {
  const layout = useContext(LayoutContext);
  return (
    <Container
      className={props.className}
      style={props.style}
      backgroundImage={props.backgroundImage}
    >
      <Content maxWidth={layout.maxWidth}>{props.children}</Content>
    </Container>
  );
}

Block.defaultProps = {
  style: {},
  className: "",
  backgroundImage: null,
};

Block.propTypes = {
  className: PropTypes.string,
  backgroundImage: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
  /**
  React CSS format to override rules has required by design
  */
  style: PropTypes.object,
};
