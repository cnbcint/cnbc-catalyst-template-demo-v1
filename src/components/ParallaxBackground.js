import React from "react";
import PropTypes from "prop-types";
import { breakpoints } from "../helpers/Layout";

import styled from "styled-components";
const Background = styled.div`
  background-image: url(${(props) => props.image.mobile});
  @media (min-width: ${breakpoints.mobile}) {
    background-image: url(${(props) => props.image.tablet});
  }
  @media (min-width: ${breakpoints.tablet}) {
    background-image: url(${(props) => props.image.desktop});
  }
  background-repeat: no-repeat;
  background-size: cover;
  background-attachment: fixed;
  background-position: center;
  position: relative;
`;
const Overlay = styled.div`
  background-color: ${(props) => props.overlayColor};
  position: absolute;
  width: 100%;
  height: 100%;
`;

const Content = styled.div`
  position: relative;
`;

export default function ParallaxBackground(props) {
  return (
    <Background
      className={props.className}
      image={props.image}
      style={props.style}
    >
      <Overlay
        overlayColor={props.overlayColor}
        className={props.className ? props.className + "__overlay" : ""}
      />
      <Content className={props.className ? props.className + "__content" : ""}>
        {props.children}
      </Content>
    </Background>
  );
}

ParallaxBackground.defaultProps = {
  overlayColor: "tranparent",
  className: "",
};

ParallaxBackground.propTypes = {
  className: PropTypes.string,

  /**
   * RGBA css property for overlay to make background do not interfere with copys
   */
  overlayColor: PropTypes.string,

  /**
   * Responsive background image this can be differen aspect ratios per screen sizes
   */
  image: PropTypes.shape({
    mobile: PropTypes.string,
    tablet: PropTypes.string,
    desktop: PropTypes.string,
  }).isRequired,
  /**
   * React CSS Overrides
   */
  style: PropTypes.object,
  /**
   * The parallax content
   */
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
};
