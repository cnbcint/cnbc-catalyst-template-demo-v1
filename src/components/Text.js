import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Pdrop = styled.p`
  color: ${(props) => props.color};
  font-size: ${(props) => (props.highlight ? "1.4em" : "1.2rem")};
  line-height: ${(props) => (props.highlight ? "140%" : "125%")};
  text-align: ${(props) => props.textAlign};
  max-width: 600px;
  margin:0 auto;
  &::first-letter {
    color: ${(props) => props.dropCapColor};
    padding: 0 0;
    margin: 0.5rem 1rem 0 0;
    font-size: 5.5rem;
    font-weight: bold;
    float: left;
    line-height: 4rem;
  }
`;
const P = styled.p`
  color: ${(props) => props.color};
  font-size: ${(props) => (props.highlight ? "1.4rem" : "1.2rem")};
  line-height: ${(props) => (props.highlight ? "140%" : "125%")};
  text-align: ${(props) => props.textAlign};
  max-width: 600px;
  margin:0 auto;

`;
/** 
 Text Block will add a paragraph element to it's children. Do not add P inside this element to avoid double nested P element
*/
export default function Text(props) {
  if (props.dropCap) {
    return (
      <Pdrop
        highlight={props.highlight}
        dropCapColor={props.dropCapColor ? props.dropCapColor : props.color}
        color={props.color}
        textAlign={props.textAlign}
        style={props.style}
        className={props.className}
      >
        {props.children}
      </Pdrop>
    );
  }
  return (
    <P
      highlight={props.highlight}
      color={props.color}
      style={props.style}
      textAlign={props.textAlign}
      className={props.className}
    >
      {props.children}
    </P>
  );
}

Text.defaulProps = {
  dropCap: false,
  color: "inherit",
  dropCapColor: "",
  highlight: false,
  className: "",
  textAlign: "left",
};

Text.propTypes = {
  /**
   Will Increase font size to highlight the text
   */
  highlight: PropTypes.bool,
  /**
   Will Increase font size of the first letter of the paragraph to achive DropCap 
   */
  dropCap: PropTypes.bool,
  /**
   CSS override for font color
   */
  color: PropTypes.string,
  /**
   CSS text aligment
   */
  textAlign: PropTypes.oneOf(["left", "right", "center"]),
  /**
   CSS color for drop cap
   */
  dropCapColor: PropTypes.string,
  /**
  React CSS format to override rules has required by design
  */
  style: PropTypes.object,
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
};
