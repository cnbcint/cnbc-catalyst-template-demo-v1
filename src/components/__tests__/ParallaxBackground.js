import React from "react";
import ParallaxBackground from "../ParallaxBackground";
import Block from "../Block";
import Text from "../Text";
import { render } from "@testing-library/react";
import "jest-styled-components";

const Component = () => (
  <ParallaxBackground
    className="test"
    overlayColor="rgba(0,0,0,.5)"
    image={{
      desktop:
        "http://localhost:8080/images/responsive/output/image03-desktop.jpg",
      tablet:
        "http://localhost:8080/images/responsive/output/image03-tablet.jpg",
      mobile:
        "http://localhost:8080/images/responsive/output/image03-mobile.jpg",
    }}
  >
    <Block>
      <Text>Testing Content</Text>
    </Block>
  </ParallaxBackground>
);

beforeAll(() => {
  window.IntersectionObserver = class {
    observe() {
      return null;
    }
    disconnect() {}
  };
});
describe("ParallaxBackground", () => {
  it("Renders correctly", () => {
    const { container } = render(<Component />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders the overlay with the correct color", () => {
    const { container } = render(<Component />);
    const target = container.getElementsByClassName("test__overlay")[0];
    expect(target).toHaveStyleRule("background-color", "rgba(0,0,0,.5)");
  });

  it("Renders the background image", () => {
    const { container } = render(<Component />);
    const target = container.getElementsByClassName("test")[0];

    expect(target).toHaveStyleRule(
      "background-image",
      "url(http://localhost:8080/images/responsive/output/image03-mobile.jpg)"
    );
  });

  it("Renders the correct content", () => {
    const { container } = render(<Component />);
    expect(
      container.getElementsByClassName("test__content")[0].textContent
    ).toMatch("Testing Content");
  });

  it("Passes the class name", () => {
    const { container } = render(<Component />);
    expect(container.firstChild.classList.contains("test")).toBe(true);
  });
});

afterAll(() => {
  window.IntersectionObserver = null;
});
