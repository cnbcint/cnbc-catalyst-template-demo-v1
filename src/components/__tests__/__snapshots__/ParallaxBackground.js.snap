// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParallaxBackground Renders correctly 1`] = `
.c0 {
  background-image: url(http://localhost:8080/images/responsive/output/image03-mobile.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-attachment: fixed;
  background-position: center;
  position: relative;
}

.c1 {
  background-color: rgba(0,0,0,.5);
  position: absolute;
  width: 100%;
  height: 100%;
}

.c2 {
  position: relative;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 3rem;
  padding-bottom: 3rem;
  background-image: none;
}

.c4 {
  width: calc(100% - 2rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 500px;
}

.c5 {
  font-size: 1.2rem;
  line-height: 125%;
}

@media (min-width:768px) {
  .c0 {
    background-image: url(http://localhost:8080/images/responsive/output/image03-tablet.jpg);
  }
}

@media (min-width:1280px) {
  .c0 {
    background-image: url(http://localhost:8080/images/responsive/output/image03-desktop.jpg);
  }
}

@media (min-width:768px) {
  .c3 {
    padding-top: 5rem;
    padding-bottom: 5rem;
  }
}

<div
  class="c0 test"
>
  <div
    class="c1 test__overlay"
  />
  <div
    class="c2 test__content"
  >
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <p
          class="c5"
        >
          Testing Content
        </p>
      </div>
    </div>
  </div>
</div>
`;
